panther_battery/src/adc_node.cpp

```diff
@@ -0,0 +1,191 @@
+#include <panther_battery/adc_node.hpp>
+
+#include <chrono>
+#include <functional>
+#include <limits>
+#include <memory>
+#include <stdexcept>
+#include <string>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <sensor_msgs/msg/battery_state.hpp>
+
+#include <panther_msgs/msg/io_state.hpp>
+
+#include <panther_battery/adc_data_reader.hpp>
+#include <panther_battery/battery_publisher.hpp>
+#include <panther_utils/moving_average.hpp>
+
+namespace panther_battery
+{
+using std::placeholders::_1;
+
+ADCNode::ADCNode(const std::string & node_name, const rclcpp::NodeOptions & options)
+: Node(node_name, options)
+{
+  this->declare_parameter<std::string>("adc0_device", "/sys/bus/iio/devices/iio:device0");
+  this->declare_parameter<std::string>("adc1_device", "/sys/bus/iio/devices/iio:device1");
+  this->declare_parameter<float>("battery_timeout", 1.0);
+  this->declare_parameter<float>("high_bat_temp", 55.0);
```

Comment:
```suggestion
  this->declare_parameter<double>("battery_timeout", 1.0);
  this->declare_parameter<double>("high_bat_temp", 55.0);
```
It is first specified as `float` and later read as `double`. This should be more consistent

Reply:
There is no method `as_float()`. However, if you declare variable as float and use `as_double()` it will cast it to float.